c++/src/Compression.hh

```diff
@@ -30,12 +30,14 @@ namespace orc {
    * @param input the input stream that is the underlying source
    * @param bufferSize the maximum size of the buffer
    * @param pool the memory pool
+   * @param metrics the reader metrics
    */
   std::unique_ptr<SeekableInputStream>
      createDecompressor(CompressionKind kind,
                         std::unique_ptr<SeekableInputStream> input,
                         uint64_t bufferSize,
-                        MemoryPool& pool);
+                        MemoryPool& pool,
+                        ReaderMetrics* metrics);
```

Comment:
If you don't mind, shall we have the default value, `ReaderMetrics* metrics = nullptr`, to be more backward-compatible?